n2survey/lime/survey.py

```diff
@@ -402,7 +421,19 @@ def plot(
 
         # Set up plot options
         theme = self.theme.copy()
-        theme.update(kwargs)
+
+        if kwargs:
+            # check if there are keywords that are not keys of theme
+            non_theme_kwargs = kwargs.copy()
+            for element in theme.keys():
+                if element in non_theme_kwargs:
+                    non_theme_kwargs.pop(element)
+            # non theme-related keywords are not yet supported
+            if non_theme_kwargs:
+                raise KeyError('the following provided kwargs are not supported:'
+                               + str(non_theme_kwargs))
+            # update the theme if there are theme-related keywords
+            theme.update(kwargs)
```

Comment:
to 1. I thought that the idea was now to raise an error as long as the plpt_functions do not handle additional keywords
to 2. Good point, so if it the keyword is 'rc', then I would have to update the 'rc' dictionary

Reply:
> to 1. I thought that the idea was now to raise an error as long as the plpt_functions do not handle additional keywords


Sorry, I did not get this. My point is that if you call `seaborn.set_theme(theme)` and `theme` contains some 'non-theme' keyword then it will raise an error. Try:

```python
theme = DEFAULT_THEME
kwargs = {"nontheme": "somevalue"}
theme.update(kwargs)
seaborn.set_theme(**theme)
```